maven-core/src/main/java/org/apache/maven/project/MavenProject.java

```diff
@@ -921,12 +926,20 @@ public void setInjectedProfileIds( String source, List<String> injectedProfileId
      * coordinates.
      *
      * @param artifact the artifact to add or replace.
-     * @throws DuplicateArtifactAttachmentException
+     * @deprecated Please use {@link MavenProjectHelper}
+     * @throws DuplicateArtifactAttachmentException not used anymore but leave it for backward compatibility
      */
     public void addAttachedArtifact( Artifact artifact )
         throws DuplicateArtifactAttachmentException
     {
-        getAttachedArtifacts().add( artifact );
+        // if already there we remove it and add again
+        int index = attachedArtifacts.indexOf( artifact );
+        if ( index > 0 )
+        {
+            LOGGER.warn( "artifact {} already attached, remove previous instance and add again" );
+            attachedArtifacts.set( index, artifact );
+        }
+        attachedArtifacts.add( artifact );
```

Comment:
do we miss an 'else' keyword here?
We are still adding the artifact twice

Reply:
argh I pushed the branch in the wrong repo

Reply:
@eolivelli fixed sorry for the confusion